### PR TITLE
Feature/filter endpoint

### PR DESCRIPTION
**Pull request by ViMuilu**
### Description

Support for search filtering.

### Related issues
issue: https://github.com/CSCfi/beacon-network-ui/issues/153

### Type of change

- [x] New feature (non-breaking change which adds functionality)

### Changes Made

Changes in query.py for querying beacon filtering term endpoints. 

Aggregator.py added /filteringTerms endpoint, this is identical to /query endpoint and was done so because only one websocket can be created per query and the /query is called when search results are queried. There is probably a smarter way of doing this.

Utils.py changes to payload processing, added filtering terms.

### Testing

All the new features use older components which have tests so i don't think that new tests are necessary.

### CODE REVIEW

**Review by blankdots — COMMENTED** [review]
i would prefer to see some tests for this, if there aren't any yet.

File: aggregator/endpoints/query.py [hunk 1]
```diff
@@ -18,15 +18,24 @@ async def send_beacon_query(request):
...
     services = await get_services(request.host)  # service urls (beacons, aggregators) to be queried
     access_token = await get_access_token(request)  # Get access token if one exists
 
     for service in services:
         # Generate task queue
-        task = asyncio.ensure_future(query_service(service, params, access_token))
-        tasks.append(task)
-
+        if "&filters=filter" in params:
+            task = asyncio.ensure_future(query_service(service, params.replace("&filters=filter", ""), access_token))
```
[the remark this answers is not in the document]
Comment (teemukataja) [reply aggregator/endpoints/query.py:32]: `asyncio.ensure_future` is used in other places too https://github.com/CSCfi/beacon-network/search?q=ensure_future

Why do we use it? Beacon was created with python 3.6, maybe there was no `create_task` back then, as that documentation suggests it is from 3.7.+

So, this is continuing the old convention, and now there are new suggestions, but the old way still works.

[the remark this answers is not in the document]
Comment (blankdots) [reply aggregator/endpoints/query.py:32]: i see, however maybe for another task we should update and refactor to support only python versions that are officially supported

[the remark this answers is not in the document]
Comment (teemukataja) [reply aggregator/endpoints/query.py:32]: Yes, good idea

**Review by blankdots — APPROVED** [review]

File: aggregator/endpoints/query.py [hunk 2]
```diff
@@ -47,16 +56,22 @@ async def send_beacon_query_websocket(request):
 
     # Task variables
     params = request.query_string  # query parameters (variant search)
+    # sets params to filter for filteringTerms to work
```
Comment (teemukataja) [inline aggregator/endpoints/query.py:59]: is this spot missing the param override which is in the sync request above?
```
    if "filteringTerms" in str(request):
        params = "filter"
```

Comment (teemukataja) [reply aggregator/endpoints/query.py:59]: After manual testing, it actually seems that this is deprecated code.

File: aggregator/endpoints/query.py [hunk 3]
```diff
@@ -18,15 +18,24 @@ async def send_beacon_query(request):
 
     # Task variables
     params = request.query_string  # query parameters (variant search)
+
+    # gets filteringterms
+    if "filteringTerms" in str(request):
+        params = "filter"
```
Comment (teemukataja) [inline aggregator/endpoints/query.py:24]:
```suggestion
```
this code seems deprecated

**Comment by teemukataja** [comment]
Added small fix from a forgotten branch to this PR https://github.com/CSCfi/beacon-network/commit/64a01f34e23feb6fcef1224aa68d47bbe57b422a

**Pull request merged**